src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7995,6 +7995,11 @@ private void ParseDeclarationModifiers(SyntaxListBuilder list)
                 {
                     mod = this.AddError(mod, ErrorCode.ERR_BadMemberFlag, mod.Text);
                 }
+                else if (list.Any(mod.Kind))
+                {
+                    // check for duplicates, can only be const
+                    mod = this.AddError(mod, ErrorCode.ERR_TypeExpected, mod.Text);
```

Comment:
Given that this function is only used to parse a local declaration, I agree that ERR_TypeExpected is in agreement with the language grammar.